pkg/command/root_test.go

```diff
@@ -908,3 +903,311 @@ func TestCompletionShortHelpInActiveHelp(t *testing.T) {
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:    "dummy",
+					target:  configtypes.TargetK8s,
+					aliases: []string{"dum"},
+				},
+				fakePluginRemapAttributes{
+					name:      "dummy2",
+					target:    configtypes.TargetK8s,
+					invokedAs: []string{"dummy3"},
+					aliases:   []string{"dum"},
+				},
+			},
+			args:     []string{"dummy3", "say", "hello"},
+			expected: []string{"hello", "the alias dum is duplicated across plugins: dummy, dummy3"},
+		},
+		{
+			test: "two plugins sharing aliases does not show duplicate warning if one replaces another",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:    "dummy",
+					target:  configtypes.TargetK8s,
+					aliases: []string{"dum"},
+				},
+				fakePluginRemapAttributes{
+					name:      "dummy2",
+					target:    configtypes.TargetK8s,
+					invokedAs: []string{"dummy"},
+					aliases:   []string{"dum"},
+				},
+			},
+			args:       []string{"dummy", "say", "hello"},
+			expected:   []string{"hello"},
+			unexpected: []string{"duplciated across plugins"},
+		},
+		{
+			test: "map to deeper subcommand is valid as long as parent exists",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:      "dummy2",
+					target:    configtypes.TargetK8s,
+					invokedAs: []string{"dummy", "kubernetes dummy"},
+				},
+			},
+			args:     []string{"kubernetes", "dummy", "say", "hello"},
+			expected: []string{"hello"},
+		},
+		{
+			test: "map to deeper subcommand is invalid if parent missing",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:      "dummy2",
+					target:    configtypes.TargetK8s,
+					invokedAs: []string{"dummy", "nonexistentprefix dummy"},
+				},
+			},
+			args:            []string{"nonexistentprefix", "dummy"},
+			expectedFailure: true,
+		},
+		{
+			test: "one mapped tmc plugin",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:      "dummy2",
+					target:    configtypes.TargetTMC,
+					invokedAs: []string{"dummy", "mission-control dummy"},
+				},
+			},
+			args:     []string{"mission-control", "dummy", "say", "hello"},
+			expected: []string{"hello"},
+		},
+		{
+			test: "two plugins remapped to same location will warn",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:      "dummy2",
+					target:    configtypes.TargetK8s,
+					invokedAs: []string{"dummy"},
+					aliases:   []string{"dum"},
+				},
+				fakePluginRemapAttributes{
+					name:      "dummy3",
+					target:    configtypes.TargetK8s,
+					invokedAs: []string{"dummy"},
+					aliases:   []string{"dum"},
+				},
+			},
+			args:     []string{"dummy", "say", "hello"},
+			expected: []string{"hello", "Warning, multiple command groups are being remapped to the same command names : \"dummy, dummy\""},
```

Comment:
Showing "dummy" twice seems not very useful.  In the code (not the test) should we instead print the plugin name so we can figure out which plugins are problematic?

Reply:
I have already added a //Todo in the code that generated the warning to  provide a better message in a followup. Just need to think through what best to show. (Note: the "dummy" twice is not reporting that two plugins is mapping to the same dummy command node, but more that two nodes name "dummy" ("dummy" and "kubernetes dummy") are both multiply mapped.